src/widgets/Step.js

```diff
@@ -0,0 +1,158 @@
+/* eslint-disable react/forbid-prop-types */
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+import React, { useState, useEffect, memo } from 'react';
+import PropTypes from 'prop-types';
+
+import Ionicons from 'react-native-vector-icons/Ionicons';
+import FontAwesome from 'react-native-vector-icons/FontAwesome';
+
+import { StyleSheet, View, Text, TouchableOpacity } from 'react-native';
+
+import { SUSSOL_ORANGE } from '../globalStyles';
+import { programStrings, modalStrings } from '../localization';
+
+const getLocalisation = ({ stepKey }) => {
+  const errorString = {
+    program: programStrings.no_programs,
+    supplier: programStrings.no_suppliers,
+    orderType: programStrings.no_order_types,
+    period: programStrings.no_periods,
+  };
+  const placeholders = {
+    program: programStrings.select_a_program,
+    supplier: programStrings.select_a_supplier,
+    orderType: programStrings.select_an_order_type,
+    period: programStrings.select_a_period,
+    name: modalStrings.give_your_stocktake_a_name,
+  };
+
+  return { placeholder: placeholders[stepKey], errorString: errorString[stepKey] };
+};
+
+/**
+ * Component rendering a sequential step in a process, according to it's
+ * current status.
+ * COMPLETE = The step is complete, render a tick and is pressable.
+ * CURRENT = The current step to complete, render an arrow and is pressable.
+ * INCOMPLETE = A future step, no icon and no pressable.
+ * ERROR = A step not able to be completed, render an X and is not pressable.
+ * @prop {Object} data - the data object the step represents. Used for displaying details
+ * @prop {String} field - The field in the data object to display
+ * @prop {String} type  - The type of step, select (dropdown) or input (text field).
+ * @prop {String} stepKey - A string key for this step.
+ * @prop {String} status - The status of this step (as above)
+ * @prop {Func}   getModalData - Callback for fetching this steps modal data
+ * @prop {Func}   onPress      - Callback for pressing this step (opens modal)
+ * @state modalData - Data to pass to the modal on opening and for determining if the
+ * current step is in an error state. Fetched when the status of this step changes,
+ * and stored for performance as to not re-fetch every re-render
```

Comment:
A few minor changes to the docstring, maybe we compromise on the whitespace and go with:

```
/**
 * Component rendering a sequential step in a process, according to it's
 * current status.
 *
 * COMPLETE = The step is complete, render a tick and is pressable.
 * CURRENT = The current step to complete, render an arrow and is pressable.
 * INCOMPLETE = A future step, no icon and no pressable.
 * ERROR = A step not able to be completed, render an X and is not pressable.
 *
 * @prop {Object} data - the data object the step represents. Used for displaying details
 * @prop {String} field - The field in the data object to display
 * @prop {String} type  - The type of step, select (dropdown) or input (text field).
 * @prop {String} stepKey - A string key for this step.
 * @prop {String} status - The status of this step (as above)
 * @prop {Func}   getModalData - Callback for fetching this steps modal data
 * @prop {Func}   onPress      - Callback for pressing this step (opens modal)
 * @state {Array} modalData - Data to pass to the modal on opening and for determining if the
 * current step is in an error state. Fetched when the status of this step changes,
 * and stored for performance as to not re-fetch every re-render
 */
```

Reply:
9c19b0916c80a178e88998822393244a60675596 cheers